### PR TITLE
EDGPATRON-22: Update raml with possible 422 http code response

### PR DESCRIPTION
**Pull request by mrdpopov**
Update raml with possible 422 http code response

### CODE REVIEW

**Comment by dcrossleyau** [comment]
It is a simple "indentation" error.

File: ramls/edge-patron.raml [hunk 1]
```diff
@@ -18,6 +18,7 @@ types:
   money: !include money.json
   item: !include item.json
   hold-cancellation: !include hold-cancellation.json
+    errors: !include raml-util/schemas/errors.schema
```
Comment (marcjohnson-kint) [inline ramls/edge-patron.raml:21]: Why is this definition indented further than the others?

Comment (oleksandr-yatsenko1) [reply ramls/edge-patron.raml:21]: It's misprint I hope so. Will fix it.

**Comment by mreno-EBSCO** [comment]
Note that the `raml` repo is not a submodule in edge-patron. So, to use "raml-util" schemas, like in this PR, either we need to create the submodule in this repo that uses the same version as the `raml` submodule in mod-patron or copy the errors/error schemas from the same version in the `raml` repo in mod-patron into this repo. Either way, the definitions need to be in sync with mod-patron. I'm probably just being pedantic here since I think the errors related schemas are fairly static at this point, so schema variation is probably not much of a concern.

There are several more places where 422 is returned via mod-patron. Check the `patron.raml` file for all the endpoints that have `is: [validate]` defined. For some reason cancel hold (request) has 422 explicitly defined instead of `is: [validate]`. This PR only addresses one 422 case.

Additionally, while we are updating the RAML file, we should update the `baseUri` to reflect that the RAML is for edge-patron and not mod-patron. It leads to a bit of confusion when looking at the API documentation for the module.

We may also want to coalesce the cancel hold API definition in the RAML so that the "view-1" version of the documentation does not have empty rows for the path segments that do not define an HTTP method, at least I assume that will fix the problem, maybe not ("view-2" seems fine):
image

mod-patron API docs have similar problems with empty rows in the "view-1" documentation, but that needs to be addressed separately.

**Comment by dcrossleyau** [comment]
That is not the correct way to add files in the shared ramls/raml-util. It is a git submodule. There are other shared files that are also needed.

You will need to remove this directory and instead add it as a git submodule. As Mathew indicates above, it may need to use the same sha pointer as does the mod-patron. 

See the guide https://dev.folio.org/guides/commence-a-module/#back-end-ramls

It seems that many "edge" modules were initially set up without this raml-util, as they probably did not need it at that time.

**Comment by oleksandr-yatsenko1** [comment]
> it may need to use the same sha pointer as does the mod-patron.

Thank you David for pointing this. I added raml module as submodule into ramls directory of edge-patron. Now it pointing to raml1.0 branch as I can see. Is that what you mean above writing that it should point on the same state of raml submodule as mod-patron does? Because on my local repo, raml-util of mod-patron pointing to "detached head" with hash "69f6074" which isn't last state of the raml module and seems it needs to be updated in mod-patron module of my local environment to point actual state.

**Comment by marcjohnson-kint** [comment]
@oleksandr-yatsenko1 

> on my local repo, raml-util of mod-patron pointing to "detached head" with hash "69f6074" which isn't last state of the raml module and seems it needs to be updated in mod-patron module of my local environment to point actual state.

I just pulled your branch, did `git submodule init` and `git submodule update`, when I navigate into the `ramls/raml-util` directory and do `git log` it reports the following as the top commit:
```
commit 4aaeffa22d5461b41ac761870dcfce58bf8db387 (HEAD, origin/raml1.0, origin/HEAD, raml1.0)
Author: Adam Dickmeiss <adam@indexdata.dk>
Date:   Thu Nov 26 17:01:59 2020 +0100

    Async tenant API FOLIO-2877 (#131)
    
    Co-authored-by: David Crossley <david@indexdata.com>
    Co-authored-by: julianladisch <julianladisch@users.noreply.github.com>
```

I think this is the latest commit and so is up to date.

Is this not what happens on your machine?

**Comment by marcjohnson-kint** [comment]
If @dcrossleyau is happy with this change, I'm happy to approve this work and get it merged

**Comment by dcrossleyau** [comment]
Please go by @mreno-EBSCO comments above, regarding which version of raml-util should be used.

https://github.com/folio-org/raml/commits/raml1.0
The version that you noted for mod-patron '69f6074' is quite old.

If it is important that edge-patron matches mod-patron, then you could move the edge-patron pointer back to that sha. Or mod-patron raml-util could be brought up-to-date.
However that decision is not for me to advise you.

Note that there is some other docs about managing the raml-util here:
https://dev.folio.org/guides/developer-setup/#update-git-submodules

**Comment by mreno-EBSCO** [comment]
I think it is OK to have version skew here since all we care about is error/errors/parameters and those files are fairly consistent across the versions of the `raml` repo.

We should apply 422 responses to the other POST APIs. Notice the 422s in mod-patron POST APIs and no 422 in the edge-patron POST APIs in the following docs:
- https://s3.amazonaws.com/foliodocs/api/mod-patron/patron.html#
- https://s3.amazonaws.com/foliodocs/api/edge-patron/edge-patron.html#

These should be consistent with each other since the edge module does very little, if any, JSON body validation and relies on mod-patron to do these checks. This is what EDGPATRON-22 was getting at.

**Review by marcjohnson-kint — APPROVED** [review]

**Pull request merged**